### PR TITLE
Support --transform option to specify "transform" plugins

### PR DESCRIPTION
**Pull request by wlupton**
Transforms are plugins and are similar in some respects to formats, but they don't have "emit" methods and don't output anything. They are collected into an "xforms" ("transforms") map (c.f. the "fmts" map) and can be specified alongside formats. The work of a transform is done in either the pre_validate_ctx() or post_validate_ctx() method. An additional setup_xform() mechod has been added (in the base class it's just an alias for setup_fmt()).

The idea is that a transform can "do stuff" to the statement tree (obviously very carefully!). Multiple transforms can be specified, and they will be run in the order that they were given on the command line.

I've written an "edit" transform that I will probably create pull request for, but it might not be general-purpose enough to be included in the pyang distribution.

(Sorry, there aren't any test cases yet... but I'll add some to the PR... I promise!)

### CODE REVIEW

**Comment by wlupton** [comment]
I created PR #437 for the demo "edit" transform.

**Comment by mbj4668** [comment]
Did you consider adding a "transform" method, similar to "emit" for normal plugins?  The idea then would be that the transform would do its work when explicitly called in transform(), rather than explicitilty in pre_/post_validate.

Apart from that, I think this looks good!

**Comment by wlupton** [comment]
I just tried to minimize changes :). I'm happy to add a transform() method. When would you call it (relative to pre_/post_validate)?

Should modules always be re-validated after a transform() method has been called? Might it make sense to allow a transform() to indicate (via its return value) whether or not re-validation is necessary?

I guess we're agreed that a transform is still a plugin, yes? Do you see any need for extending the plugin class hierarchy to distinguish transform plugins and output formats?

**Comment by wlupton** [comment]
I've done what you suggest (_much_ better!). I made some executive decisions about when to call it and how to handle the return value (I figured it was necessary to revalidate after each transform, because the next transform might rely on this).

File: pyang/plugin.py [hunk 1]
```diff
@@ -117,6 +129,9 @@ def setup_fmt(self, ctx):
         """
         return
 
+    # setup_xform is preferred to setup_fmt for transform plugins
+    setup_xform = setup_fmt
```
Comment (mbj4668) [inline pyang/plugin.py]: I think setup_xform MUST be used by transform plugins.  I.e., define setup_xform as a separate method.  No reason to accept a transform plugin to be able to use setup_fmt.

Comment (wlupton) [reply pyang/plugin.py]: Agreed. I've committed and pushed a fix.

**Comment by mbj4668** [comment]
Thanks for this contribution!

**Pull request merged**